test/extended/jenkins/plugin.go

```diff
@@ -13,185 +11,441 @@ import (
+			envs := []kapi.EnvVar{}
+			if build.Spec.Strategy.DockerStrategy != nil && build.Spec.Strategy.DockerStrategy.Env != nil {
+				envs = build.Spec.Strategy.DockerStrategy.Env
+			} else if build.Spec.Strategy.SourceStrategy != nil && build.Spec.Strategy.SourceStrategy.Env != nil {
+				envs = build.Spec.Strategy.SourceStrategy.Env
+			} else {
+				continue
+			}
+
+			for k, v := range varsToFind {
+				found := false
+				for _, env := range envs {
+					ginkgolog("Found %s=%s in build %s", env.Name, env.Value, build.GetName())
+					if k == env.Name && v == env.Value {
+						found = true
+						break
+					}
+				}
+				o.ExpectWithOffset(1, found).To(o.BeTrue())
+			}
+		}
+	}
+}
+
+func getAdminPassword(oc *exutil.CLI) string {
+	envs, err := oc.Run("set").Args("env", "dc/jenkins", "--list").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	kvs := strings.Split(envs, "\n")
+	for _, kv := range kvs {
+		if strings.HasPrefix(kv, "JENKINS_PASSWORD=") {
+			s := strings.Split(kv, "=")
+			fmt.Fprintf(g.GinkgoWriter, "\nJenkins admin password %s\n", s[1])
+			return s[1]
+		}
+	}
+	return ""
 }
 
 var _ = g.Describe("[jenkins][Slow] openshift pipeline plugin", func() {
 	defer g.GinkgoRecover()
 	var oc = exutil.NewCLI("jenkins-plugin", exutil.KubeConfigPath())
-	var hostPort string
-	var password string
+	var j *JenkinsRef
 
 	g.BeforeEach(func() {
+		testNamespace := oc.Namespace()
 
-		g.By("set up policy for jenkins jobs")
-		err := oc.Run("policy").Args("add-role-to-user", "edit", "system:serviceaccount:"+oc.Namespace()+":default").Execute()
-		o.Expect(err).NotTo(o.HaveOccurred())
+		// If Jenkins has already been initialized, we are done setting up
+		if j == nil {
+			jenkinsNamespace := oc.Namespace() + "-jenkins"
+			g.By("Starting a Jenkins instance in namespace: " + jenkinsNamespace)
 
-		g.By("kick off the build for the jenkins ephermeral and application templates")
-		tag := []string{"openshift/jenkins-plugin-snapshot-test:latest"}
-		hexIDs, err := exutil.DumpAndReturnTagging(tag)
-		var jenkinsEphemeralPath string
-		var testingSnapshot bool
-		if len(hexIDs) > 0 && err == nil {
-			// found an openshift pipeline plugin test image, must be testing a proposed change to the plugin
-			jenkinsEphemeralPath = exutil.FixturePath("testdata", "jenkins-ephemeral-template-test-new-plugin.json")
-			testingSnapshot = true
-		} else {
-			// no test image, testing the base jenkins image with the current, supported version of the plugin
-			jenkinsEphemeralPath = exutil.FixturePath("..", "..", "examples", "jenkins", "jenkins-ephemeral-template.json")
-		}
-		err = oc.Run("new-app").Args(jenkinsEphemeralPath).Execute()
-		o.Expect(err).NotTo(o.HaveOccurred())
-		jenkinsApplicationPath := exutil.FixturePath("..", "..", "examples", "jenkins", "application-template.json")
-		err = oc.Run("new-app").Args(jenkinsApplicationPath).Execute()
-		o.Expect(err).NotTo(o.HaveOccurred())
+			oc.Run("new-project").Args(jenkinsNamespace).Execute()
```

Comment:
I've now convinced myself that there is no clean/reliable way to tear down the namespace in Ginkgo. The framework intentionally disallows discourages this behavior. There is even a comment in their repo about how it would help k8s testing: https://github.com/onsi/ginkgo/issues/70#issuecomment-226818957

That leaves:
(1) Using Ben's new numeric template parameter, allow template to use ephemeral/unique port for the Jenkins server and leave it up for all tests.
(2) Bring up Jenkins on each test.
(3) Put all tests into the same spec.

I'm personally a fan of (2) at this point, so we can keep the specs clean.